### PR TITLE
feat: add model pull api

### PR DESCRIPTION
**Pull request by namchuai**
## Describe Your Changes

- CLI command to pull model: `build/cortex.cpp models pull llama3`
- API to pull model: 
```
curl --location 'http://localhost:3928/models/pull' \
--header 'Accept-Language: en-US' \
--header 'Connection: keep-alive' \
--header 'Referer: http://localhost:3000/' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-site' \
--header 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Jan/0.1.3 Chrome/124.0.6367.207 Electron/30.0.6 Safari/537.36' \
--header 'accept: application/json' \
--header 'authorization: Bearer' \
--header 'content-type: application/json' \
--header 'sec-ch-ua: "Not-A.Brand";v="99", "Chromium";v="124"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "macOS"' \
--header 'x-stainless-arch: unknown' \
--header 'x-stainless-lang: js' \
--header 'x-stainless-os: Unknown' \
--header 'x-stainless-package-version: 0.0.4' \
--header 'x-stainless-runtime: browser:chrome' \
--header 'x-stainless-runtime-version: 124.0.6367' \
--data '{
  "modelId": "tinyllama"
}'
```

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

**Review by nguyenhoangthuan99 — APPROVED** [review]

File: controllers/models.cc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include "models.h"
+
+void Models::PullModel(const HttpRequestPtr& req,
+                       std::function<void(const HttpResponsePtr&)>&& callback) {
+  auto modelHandle = (*(req->getJsonObject())).get("modelId", "").asString();
```
Comment (vansangpfiev) [inline controllers/models.cc:5]: not sure if we should use `model_id` or `modelId`

Comment (vansangpfiev) [inline controllers/models.cc:5]: JsonObject can be null, maybe we should check before using it

File: services/download_service.cc [hunk 2]
```diff
@@ -2,17 +2,27 @@
...
 
   for (const auto& item : task.items) {
     StartDownloadItem(task.id, item);
   }
 }
 
+void DownloadService::AddAsyncDownloadTask(const DownloadTask& task) {
+  tasks.push_back(task);
+  for (const auto& item : task.items) {
+    std::thread([this, task, item]() {
```
Comment (vansangpfiev) [inline services/download_service.cc:20]: I'm not sure if .detach() will be safe, please add a note here to try other APIs like https://en.cppreference.com/w/cpp/thread/async later.

File: services/download_service.cc [hunk 3]
```diff
@@ -2,17 +2,27 @@
...
+      this->StartDownloadItem(task.id, item);
+    }).detach();
+  }
+}
+
 const std::string DownloadService::GetContainerFolderPath(DownloadType type) {
   std::filesystem::path containerFolderPath;
```
Comment (vansangpfiev) [inline services/download_service.cc:27]: nitpick: naming convention, should be container_folder_path

**Review by vansangpfiev — APPROVED** [review]

File: controllers/models.cc (hunk 1 above)
Comment (namchuai) [reply controllers/models.cc:5]: I think let's keep it as it is for now. We can change it later on.

File: services/download_service.cc (hunk 2 above)
Comment (namchuai) [reply services/download_service.cc:21]: added a todo there for future improvement
